4-Express/client/scripts/todo_list.js

```diff
@@ -0,0 +1,241 @@
+  const newItemInput = document.getElementById("new-item-name");
+  const addBtn = document.getElementById("add-btn");
+
+  function createButton(type, cb) {
+    const button = document.createElement("div");
+    const icon = iconMap[type] || "&#xe145;";
+
+    const iconSpan = document.createElement("span");
+    iconSpan.classList.add("material-icons", "md-36");
+    iconSpan.innerHTML = icon;
+
+    button.classList.add("btn", classes[type + "Btn"]);
+    button.dataset.type = type;
+    button.appendChild(iconSpan);
+    button.addEventListener("click", cb);
+
+    return button;
+  }
+
+  function createItemLabel(id, text) {
+    const label = document.createElement("label");
+    label.classList.add(classes.item);
+    label.dataset.type = TYPE_ITEM_LABEL;
+    label.innerText = text;
+    label.addEventListener("dblclick", editItemCallback(id));
+
+    return label;
+  }
+
+  function createItemInput(id, text) {
+    const input = document.createElement("input");
+    input.type = "text";
+    input.classList.add("edit");
+    input.dataset.type = TYPE_ITEM_INPUT;
+    input.value = text;
+    input.addEventListener("keyup", confirmItemEditCallback(id));
+
+    return input;
+  }
+
+  function createTodoItem(id, text, check) {
+    const newItemContainer = document.createElement("div");
+    newItemContainer.classList.add("bar", "item-bar");
+    newItemContainer.dataset.id = id;
+
+    const newItemCheckBtn = createButton(TYPE_CHECK_BTN, checkItemCallback(id));
+    newItemContainer.appendChild(newItemCheckBtn);
+
+    if (check) {
+      newItemCheckBtn.dataset.checked = "true";
+      newItemCheckBtn.classList.add("checked");
+    } else {
+      newItemCheckBtn.dataset.checked = "false";
+    }
+
+    const newItemLabel = createItemLabel(id, text);
+    newItemContainer.appendChild(newItemLabel);
+
+    const newItemEditBtn = createButton(TYPE_EDIT_BTN, editItemCallback(id));
+    newItemContainer.appendChild(newItemEditBtn);
+
+    const newItemRemoveBtn = createButton(
+      TYPE_REMOVE_BTN,
+      removeItemCallback(id)
+    );
+    newItemContainer.appendChild(newItemRemoveBtn);
+
+    return newItemContainer;
+  }
+
+  function addItemCallback(event) {
+    if (
+      event.type === "click" ||
+      (event.type === "keyup" && event.keyCode === KEY_CODE_ENTER)
+    ) {
+      if (isVaildItemName(newItemInput.value)) {
+        const uniqueId = uuidv4();
+        const newItem = createTodoItem(uniqueId, newItemInput.value, false);
+
+        document.getElementById("main-container").appendChild(newItem);
+        saveToServer(uniqueId);
+
+        newItemInput.value = "";
+      }
+    }
+  }
+
+  function isVaildItemName(name) {
+    let valid = name.length > 0;
+
+    if (!valid) {
+      alert("Invalid item name!");
+    }
+
+    return valid;
+  }
+
+  function getItemElement(id, type) {
+    let element = document.querySelector("[data-id='" + id + "']");
```

Comment:
string interpolation can fit well here
```suggestion
    let element = document.querySelector(`[data-id='"${id}"']`);
```